### ISSUE TITLE
hostgroup static - │ Error: Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by shiransiboni** [CrowdStrike/terraform-provider-crowdstrike#36]
I notice you changed the logic for static hostgroup and add the options for hostname and host_ids parameters 
I got this error - 

This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.

![image](https://github.com/user-attachments/assets/a40a3b38-a9d7-43ff-9569-03eab8732bc5)

**Comment by ffalor** [comment]
@shiransiboni can you verify in the console which group type this host group is?

If in the console the group type is `Static by hostname` then the group type should be `static` in terraform. If the console says `Static by host ID` then the group type should be `staticByID` in terraform.

If those are correct could you provide me your config and the `terraform plan` details for this resource? With any sensitive information redacted please.

Thanks!

**Comment by shiransiboni** [comment]
This is what I did  already
![image](https://github.com/user-attachments/assets/d1345b56-cf55-4fb3-9929-0831ace8c19b)

 but its broken in the "apply" process (passed the plan and in the apply Im getting the error: 
![Screenshot 2024-08-04 at 10 13 52](https://github.com/user-attachments/assets/26aa4e0b-b5fe-4d4e-801c-d1343cf5e5bb)

**Comment by shiransiboni** [comment]
I found how to fixed it for now, but I think you should have a look -
I can't do any modification except deletion - So, I removed all the host groups - apply it and then create a new process and all works.

**Comment by shiransiboni** [comment]
When I created a prevention policy, I encountered a critical issue. For example, when I changed the policy setting from "false" to "true," I received an error. However, the UI shows that the changes have been applied, but the pull request remains open due to the error.

The issue is resolved when I run the pipeline, delete the policy, and then reapply it with the changes. However, I cannot proceed until this bug is fixed.

**Comment by ffalor** [comment]
@shiransiboni originally were these host groups imported? The error makes me thing you had the type as `staticByID` but these were actually `static` host group types which is why the error is given. This would also be the reason why deleting and recreating would seem to "fix" the issue because now the group type in the code matches the group type in the console.

Does this seem right to you?

**Comment by shiransiboni** [comment]
No.

This is a new group that I created using a Terraform pipeline. After applying it, everything worked. However, after you made changes and added the option for host_ids, I updated the configuration to include this filter. When I changed from static to staticByID, I encountered this error.

If I delete everything and create it again with the changes, it will work. But that's not the situation. I need to make modifications without starting from scratch; that's the whole point.

**Comment by ffalor** [comment]
> When I changed from static to staticByID, I encountered this error.
Correct group type can't be changed without a delete and recreate. Once a host group is created the type can't be changed. I'll update the attributes so this will be shown in the plan to prevent you from having to do a delete and re-create in separate steps.

All other host group changes are "in-place updates", but group type requires a delete & recreate to change after a host group is created.

**Comment by shiransiboni** [comment]
but I dont want to changed the host type here - now I want to change the prevention policy to enable and still get it
so want should I need to do?

**Comment by ffalor** [comment]
Hey @shiransiboni the original issue reported in this thread was about host groups. I've identified a fix based on your feedback. I do see your mention of prevention policy, but I am unsure how it relates to the host group. Is the prevention policy related to this issue, or is it related to your other github issue #35, or are you seeing something else?

Sorry for the back & forth.. thanks for reporting these issues and giving me the opportunity to fix them.

**Comment by ffalor** [comment]
#40 will do a replace when the host group type in the terraform config does not match what the console is reporting. This will prevent the need to delete the resource before recreating it.

**Comment by ffalor** [comment]
Fixed in version `v0.0.7`

```terraform
terraform {
  required_providers {
    crowdstrike = {
      source = "CrowdStrike/crowdstrike"
      version = "0.0.7"
    }
  }
}
```
